### PR TITLE
Alerting: Add tests for Slack Notifier Image Upload

### PR DESCRIPTION
**Pull request by joeblubaugh**
Tests that file bytes are correctly read from storage when an image is
available without a URL. Tests that not found and success cases are both
handled.

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/slack_test.go [hunk 1]
```diff
@@ -271,6 +412,12 @@ func TestSlackNotifier(t *testing.T) {
 			require.NoError(t, err)
 
 			require.JSONEq(t, string(expBody), body)
+
+			if c.webhookSent {
+				require.True(t, len(notificationService.Webhook.Url) > 0)
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/channels/slack_test.go:417]: In this case we're asserting that we tried to upload the image file, correct? Should we compare the URL against the actual expected URL for this endpoint? This is less brittle than it seems if we use `Contains` and compare against a constant, like `require.Contains(t, notificationService.Webhook.Url, slackUploadImageUrl)` maybe?

**Review by alexweav — APPROVED** [review]
LGTM

File: pkg/services/ngalert/notifier/channels/slack_test.go (hunk 1 above)
Comment (joeblubaugh) [reply pkg/services/ngalert/notifier/channels/slack_test.go:417]: I like it

Comment (joeblubaugh) [reply pkg/services/ngalert/notifier/channels/slack_test.go]: I ended up injecting the expected URL so we could test the custom image domain case too.

**Pull request merged**